src/shared/components/oncoprint/ResultsViewOncoprintUtils.ts

```diff
@@ -0,0 +1,87 @@
+import {CoverageInformation} from "../../../pages/resultsView/ResultsViewPageStoreUtils";
+import {MolecularProfile} from "../../api/generated/CBioPortalAPI";
+import {OncoprintClinicalAttribute} from "./ResultsViewOncoprint";
+import {SpecialAttribute} from "../../cache/OncoprintClinicalDataCache";
+import _ from "lodash";
+import naturalSort from "javascript-natural-sort";
+
+export const alterationTypeToProfiledForText:{[alterationType:string]:string} = {
+    "MUTATION_EXTENDED": "mutations",
+    "COPY_NUMBER_ALTERATION": "copy number alterations",
+    "MRNA_EXPRESSION": "mRNA expression",
+    "PROTEIN_LEVEL": "protein expression"
+};
+
+export function makeProfiledInClinicalAttributes(
+    coverageInformation: CoverageInformation["samples"],
+    molecularProfileIdToMolecularProfile: {[molecularProfileId:string]:MolecularProfile},
+    selectedMolecularProfiles: MolecularProfile[],
+    isSingleStudyQuery: boolean
+) {
+    // determine which Profiled In clinical attributes will exist in this query.
+    // A Profiled In <alteration type> attribute only exists if theres a sample in the query
+    //  which is not profiled in any selected profiles for that type.
+
+    // If its a single study query (isSingleStudyQuery), and there is only one profile of a particular alteration type
+    //  in the study, then an attribute will be generated specifically for that profile. Otherwise, the attribute
+    //  will represent an entire alteration type.
+
+    const groupedSelectedMolecularProfiles:{[alterationType:string]:MolecularProfile[]} =
+        _.groupBy(selectedMolecularProfiles, "molecularAlterationType");
+    const selectedMolecularProfilesMap = _.keyBy(selectedMolecularProfiles, p=>p.molecularProfileId);
+
+    const existsUnprofiled:{[alterationType:string]:boolean} = _.reduce(coverageInformation, (map, sampleCoverage)=>{
```

Comment:
i believe if you type the argument "map" in the reducer callback, you can avoid having to to type both the variable and then cast the initial memo (third arg of reduce)